### PR TITLE
refactor: share API error normalization

### PR DESCRIPTION
**Pull request by jan-kubica**
## Summary

- normalize HTTP and Eden error payloads in @stll/api-contract
- keep presentation and localization in the web adapter
- preserve raw messages and structured details with exact optional semantics

## Validation

- bun --filter @stll/api-contract typecheck
- bun --filter @stll/api-contract lint
- bun --filter @stll/api-contract test
- VITE_API_URL=http://localhost:3001 bun test apps/web/src/lib/errors/index.test.ts
- bun packages/scripts/src/tsc-native.ts --noEmit -p apps/web --pretty false
- pre-push affected-workspace formatting, localization, and typecheck gates

## Summary by CodeRabbit

* **New Features**
  * Improved API error normalisation for null/undefined, text messages, validation-shaped errors, and structured error objects.
  * Added shared exported utilities and types to keep error shapes consistent across the app.

* **Bug Fixes**
  * Updated error conversion to preserve message/details consistently and to localise output reliably.

* **Tests**
  * Added Bun test coverage for API error normalisation behaviours, including edge cases like `value: undefined`.

### CODE REVIEW

File: packages/api-contract/src/error.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { describe, expect, test } from "bun:test";
+
+import { API_VALIDATION_ERROR_CODE, normalizeApiError } from "./error";
+
+describe("normalizeApiError", () => {
+  test("keeps an empty response status-only", () => {
+    expect(normalizeApiError({ status: 502, value: null })).toEqual({
+      status: 502,
+    });
+  });
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply packages/api-contract/src/error.test.ts:13]: Added the explicit undefined assertion in 69b2ef94b. The focused contract tests, lint, and formatting check all pass.

**Comment by jan-kubica** [comment]
Addressed both CodeRabbit suggestions: explicit undefined-body coverage in 69b2ef94b and a single validation discriminator source in 6e385be7a. The api-contract typecheck, tests, lint, and formatting checks pass.
